### PR TITLE
Add draft for File input

### PR DESCRIPTION
**Pull request by mrmckeb**
Relates to PR #92, issue #114.

### CODE REVIEW

**Comment by yoavweiss** [comment]
@mrmckeb could you join the WICG to appease the IPR bots?

**Comment by mrmckeb** [comment]
Hi @yoavweiss, thanks for the note. I did - after creating this PR - but it doesn't seem to have updated... maybe the bot can be re-run?

**Comment by yoavweiss** [comment]
Re-ran the bots and they seem to like you now :) Welcome!!

**Comment by mrmckeb** [comment]
The stars have aligned, thanks @yoavweiss!

**Comment by gregwhitworth** [comment]
ping @mrmckeb any update on this?

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:176]: I've updated this, and also raised #173.

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:206]: OK, thanks. I've raised #174.

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:224]: I think it's valuable to understand that, today, this is the common workaround - but it doesn't add value to _this_ document as a proposal.

I've raised #175. Let's see what others think, and I can add to or remove this section based on that discussion.

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:159]: By `open`, do you mean that the system's file picker is open?

If so, this is not something I've seen in anything I've come across during research. Possibly because it's not possible to do (if you're not a browser vendor), and because the user is already _trapped_ in the system's file picker until they've selected a file.

File: research/src/pages/file/file.proposal.mdx
Comment (gregwhitworth) [inline research/src/pages/file/file.proposal.mdx:80]: Then let's not include it.

File: research/src/pages/file/file.proposal.mdx
Comment (gregwhitworth) [reply research/src/pages/file/file.proposal.mdx:126]: yes, this is for when the window is open just in case you want to do some variant. Given we are adding the select has an open state that the popup is open seems for consistency reasons to do it here. You're correct in the current world since no file system access that the doc will be frozen by the modal but that doesn't mean that will always be the case. But this can be discussed in an issue rather than plague your PR :)

File: research/src/components/file-anatomy.js
Comment (gregwhitworth) [inline research/src/components/file-anatomy.js:11]: Can you file an issue about the naming of this? I think we should discuss having this simply be represented by `<file>` rather than input. This won't block the PR however since the current anatomy has been approved.

**Review by gregwhitworth — APPROVED** [review]
I left some feedback on some minor tweeks and an additional issue or two for further discussion. Thanks for taking the time on this - it looks good.

File: research/src/components/file-anatomy.js
Comment (mrmckeb) [reply research/src/components/file-anatomy.js:11]: Done! https://github.com/WICG/open-ui/issues/180

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:80]: Updated.

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:126]: I've raised an issue, what type of research do you have in mind here? I wasn't sure what to add.
https://github.com/WICG/open-ui/issues/181

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:40]: Done!

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:56]: I've added the GitHub one and will try to find some more examples to add.

File: research/src/pages/file/file.proposal.mdx
Comment (gregwhitworth) [reply research/src/pages/file/file.proposal.mdx:123]: @mrmckeb research would be is there a common paradigm that when a file popup is open that this has implications on other aspects of the scenario? EG: Do icons, style or content change in any meaningful way?

**Comment by gregwhitworth** [comment]
@mrmckeb gentle ping, any update?

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx:123]: As the browser doesn't provide an open state, custom components can't do anything to show this state change.

Native file inputs also provide no visual feedback that the picker is open - apart from the picker itself, which blocks further interaction with the page until closed.

File: research/src/components/file-anatomy.js
Comment (gregwhitworth) [inline research/src/components/file-anatomy.js:12]: somehow I completely missed this - but this part should be:

`file-selector-button`

Per this resolution in relation with the CSSWG: https://github.com/WICG/open-ui/issues/91#issuecomment-639160328

File: research/src/pages/file/file.proposal.mdx
Comment (gregwhitworth) [inline research/src/pages/file/file.proposal.mdx:143]: can you file an issue as this should be more clear on what "keyboard navigation" means. Likewise we'll want to understand if there are any name computation adjustments that should be made (eg: on change, etc)

File: research/src/pages/file/file.proposal.mdx
Comment (gregwhitworth) [inline research/src/pages/file/file.proposal.mdx:150]: Can we open an issue for this and link to it?

File: research/src/pages/file/file.proposal.mdx
Comment (gregwhitworth) [inline research/src/pages/file/file.proposal.mdx:162]: adjust this to the ARIA role of `button`.

Also, this shouldn't be referencing the "control" please use the anatomy part that you've defined as the host, so in this case oui-file.

File: research/src/components/file-anatomy.js
Comment (gregwhitworth) [inline research/src/components/file-anatomy.js:11]: we agreed in a telecon that we should have the Open UI components be that of what we would propose as the new element but under a web component namespace.

So let's change this to `oui-file`.

**Review by gregwhitworth — CHANGES_REQUESTED** [review]
Sorry for finding more changes, I apologize. I would like to land this. @una can you give a review of this as well so that @mrmckeb can possibly only do 1 more pass and we can land the PR.

**Comment by mrmckeb** [comment]
Thanks @gregwhitworth, I'll wait for @una's review and then make the final changes! Looking forward to getting this in :)

File: research/src/components/file-anatomy.js
Comment (mrmckeb) [reply research/src/components/file-anatomy.js]: Done, it's now `openui-file`.

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx]: Updated, thanks!

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx]: Perhaps we drop this statement? Really this just means that the input has is tabbable, and I wouldn't expect it to blur on change (aligning with other controls).

File: research/src/pages/file/file.proposal.mdx
Comment (mrmckeb) [reply research/src/pages/file/file.proposal.mdx]: Done, thanks.
https://github.com/WICG/open-ui/issues/248

File: research/src/pages/file/file.proposal.mdx
Comment (gregwhitworth) [inline research/src/pages/file/file.proposal.mdx:2]: This should be proposals - although I do have a desire to move this to a different name in the future. However let's not change it in this PR.

File: research/src/pages/file/file.research.mdx
Comment (gregwhitworth) [inline research/src/pages/file/file.research.mdx:3]: Please put this under the analysis menu and rename it to File only

**Review by gregwhitworth — CHANGES_REQUESTED** [review]
There are some errors in the console that relate to the anatomy. Can you please ensure that those are addressed. We're almost there, only a few small changes. Thank you again for your work on this.

**Comment by mrmckeb** [comment]
Thanks @gregwhitworth, this is updated, and squashed.

File: research/src/pages/file/file.proposal.mdx
Comment (gregwhitworth) [reply research/src/pages/file/file.proposal.mdx]: Let's not block the PR on this, but I realized this was already resolved in here: https://github.com/WICG/open-ui/issues/107

**Review by gregwhitworth — APPROVED** [review]
Please see the comment regarding the resolution of the tabbing behavior for #248. I don't want to block that edit on this PR but if you can submit a new PR once this one lands with that resolved change.

**Comment by mrmckeb** [comment]
Thanks @gregwhitworth - I can make that change as soon as this lands.

**Comment by gregwhitworth** [comment]
@mrmckeb this is approved by me so we need to get one more to land it. I actually have a triage session with @una tomorrow so I'll make sure bring this to her attention.

**Comment by una** [comment]
LGTM, nice breakdown

**Pull request merged**
